### PR TITLE
[Enhancement] support table name with one leading underscore

### LINKED ISSUE #11640: support table name with one underscore
## Enhancement

Support table name with one underscore. Table name with more than one consecutive underscores are not allowed.

```
create table _test...  # ok
create table __test.. # not ok
```

### PR DESCRIPTION
**Pull request by padmejin**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #11640 

## Problem Summary(Required) ：

Modify the regular expression of the table name so that table name with one underscore is considered legal and table name with more than one consecutive underscores is not.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] I have added user document for my new feature or new function

### CODE REVIEW

**Comment by padmejin** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: fe/fe-core/src/main/java/com/starrocks/common/FeNameFormat.java [hunk 1]
```diff
@@ -27,15 +27,17 @@
 import com.starrocks.system.SystemInfoService;
 
 public class FeNameFormat {
-    private static final String LABEL_REGEX = "^[-_A-Za-z0-9]{1,128}$";
-    public static final String COMMON_NAME_REGEX = "^[a-zA-Z][a-zA-Z0-9_]{0,63}$";
+    private FeNameFormat() {}
+
+    private static final String LABEL_REGEX = "^[-\\w]{1,128}$";
+    public static final String COMMON_NAME_REGEX = "^[a-zA-Z]\\w{0,63}$|^_[a-zA-Z0-9]\\w{0,62}$";
```
Comment (nshangyiming) [inline fe/fe-core/src/main/java/com/starrocks/common/FeNameFormat.java:33]: You mean the leading underscore format is newly supported? And why multi leading underscores format is not supported, is there any special reason?

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/common/FeNameFormat.java:33]: This feature is required by Celonis.

Comment (nshangyiming) [reply fe/fe-core/src/main/java/com/starrocks/common/FeNameFormat.java:33]: got it , suggested title: [Enhancement] support table name with one leading underscore

Comment (padmejin) [reply fe/fe-core/src/main/java/com/starrocks/common/FeNameFormat.java:33]: done

**Review by gengjun-git — APPROVED** [review]

**Review by nshangyiming — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by padmejin** [comment]
run starrocks_admit_test

**Comment by padmejin** [comment]
@Mergifyio backport branch-2.4 branch-2.3 branch-2.2
¡Muchas gracias!

**Pull request merged**